### ISSUE TITLE
1.8.0 failed to load cuda shared library

### ISSUE THREAD

**Issue by SachaWalton** [microsoft/onnxruntime#8177]
Hello,
I am updating from onnxruntime 1.7.1 to 1.8.0. The script utilises the CUDA execution provider. No other code changes have been made between updates. However following error is reported:

> 2021-06-28 18:09:32.6703074 [E:onnxruntime:test, provider_bridge_ort.cc:902 onnxruntime::ProviderSharedLibrary::Ensure] Failed to load library, error code: 126
> ORT failure OrtSessionOptionsAppendExecutionProvider_Cuda: Failed to load shared library

The execution then falls back to CPU and successfully completes a forward pass of the models.

**System information**
- OS Platform and Distribution: Windows 10 2004
- ONNX Runtime installed from (source or binary): NuGet
- ONNX Runtime version: 1.8.0
- Visual Studio version (if applicable): 15.9.35
- CUDA/cuDNN version: 11.0.3 / 8.0.2.39
- GPU model and memory: Nvidia RTX 2070 

**To Reproduce**
```
OrtSessionOptions *options_;
OrtSession *session_;

int deviceID_;
int memory_limit_;

OrtApiWrapper::CheckStatus(OrtApiWrapper::instance().get()->CreateSessionOptions(&options_));
		OrtApiWrapper::CheckStatus(OrtApiWrapper::instance().get()->SetSessionGraphOptimizationLevel(options_, ORT_ENABLE_ALL));

			OrtCUDAProviderOptions cuda_options{
				deviceID_, // device id
				static_cast<OrtCudnnConvAlgoSearch>(EXHAUSTIVE),
				static_cast<int>(memory_limit_ * 1024 * 1024), // cuda_mem_limit
				1, // arena_extend_strategy
				1, // do_copy_in_default_stream
				0,
				nullptr };
			OrtApiWrapper::CheckStatus(OrtApiWrapper::instance().get()->SessionOptionsAppendExecutionProvider_CUDA(options_, &cuda_options));
```

Thanks,
Sacha

**Comment by hariharans29** [comment]
In ORT 1.8.0, the CUDA provider is now a shared library. The shared provider dlls weren't bin placed to the binary folder in 1.8.0. This has been resolved with https://github.com/microsoft/onnxruntime/pull/8138. Can you check if manually copying over the shared provider dlls from the Nuget to the binary folder resolves the issue for you ?

**Comment by SachaWalton** [comment]
Thank you, this has resolved my issue.

The files "onnxruntime_providers_cuda.dll" and "onnxruntime_providers_shared.dll" were missing from the build. 
Manually copying these from "packages\Microsoft.ML.OnnxRuntime.Gpu.1.8.0\runtimes\win-x64\native" to the build directory resolved the issue.

**Comment by hariharans29** [comment]
Thanks for confirming. We added logic to automate the copying of theses dlls to the build directory (just like onnxruntime.dll is being copied over automatically) and this should be available in 1.8.1

**Comment by z1521183003** [comment]
Hello, I am using version 1.11.0 now, but there are still problems

![image](https://user-images.githubusercontent.com/29905356/177063401-3b3b1cb5-c112-446f-a1ad-3731183eb06c.png)


![image](https://user-images.githubusercontent.com/29905356/177063310-9b7a40cb-adb2-4c79-8ddc-2a09c864b88f.png)


earliest question


![image](https://user-images.githubusercontent.com/29905356/177063524-00b5519c-935a-447a-815d-883e6575217d.png)

So I added OrtEnv.Instance();
but now
![image](https://user-images.githubusercontent.com/29905356/177063692-6b2f21f0-7e38-4164-b9dd-d2e526fe6206.png)


I have these dlls in my build directory
![image](https://user-images.githubusercontent.com/29905356/177063742-a084a519-dbcd-4a55-8cdf-f3396019454a.png)
